### PR TITLE
Fix per-tenant silence limits not reloaded during runtime

### PR DESCRIPTION
**Pull request by grobinson-grafana**
#### What this PR does

This commit fixes a bug where per-tenant silence limits were not reloaded during runtime. This happened because per-tenant limits were read at initialization time for a per-tenant Alertmanager and then never again. Instead, silence limits are read each time they are needed. Mimir uses an in-memory cache for per-tenant limits, and silence limits are only checked when creating new silences or updating existing silences.

#### Which issue(s) this PR fixes or relates to

Fixes https://github.com/grafana/alerting-squad/issues/825

#### Checklist

- [x] Tests updated.
- [x] Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [x] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

File: go.mod
Comment (grobinson-grafana) [inline go.mod:295]: Needs to be changed as soon as https://github.com/grafana/prometheus-alertmanager/pull/87 is merged.

**Review by gotjosh — APPROVED** [review]
This LGTM, but you should definitely wait for a review from the @grafana/alerting-squad

**Comment by grobinson-grafana** [comment]
@grafana/alerting-squad please take a look! I'll merge this on Thursday so it makes next week's dev release.

**Review by stevesg — APPROVED** [review]

**Pull request merged**
